plugins/csi/plugin.go

```diff
@@ -148,6 +154,69 @@ func (r *NodePublishVolumeRequest) Validate() error {
 	return nil
 }
 
+type NodeStageVolumeRequest struct {
+	// The external ID of the volume to stage.
+	ExternalID string
+
+	// If the volume was attached via a call to `ControllerPublishVolume` then
+	// we need to provide the returned PublishContext here.
+	PublishContext map[string]string
+
+	// The path to which the volume MAY be staged. It MUST be an
+	// absolute path in the root filesystem of the process serving this
+	// request, and MUST be a directory. The CO SHALL ensure that there
+	// is only one `staging_target_path` per volume. The CO SHALL ensure
+	// that the path is directory and that the process serving the
+	// request has `read` and `write` permission to that directory. The
+	// CO SHALL be responsible for creating the directory if it does not
+	// exist.
+	// This is a REQUIRED field.
+	StagingTargetPath string
+
+	// Volume capability describing how the CO intends to use this volume.
+	VolumeCapability *VolumeCapability
+
+	// Secrets required by plugins to complete the node stage volume
+	// request. This field is OPTIONAL.
+	Secrets structs.CSISecrets
+
+	// Volume context as returned by SP in the CSI
+	// CreateVolumeResponse.Volume.volume_context which we don't implement but
+	// can be entered by hand in the volume spec.  This field is OPTIONAL.
+	VolumeContext map[string]string
+}
+
+func (r *NodeStageVolumeRequest) ToCSIRepresentation() *csipbv1.NodeStageVolumeRequest {
+	if r == nil {
+		return nil
+	}
+
+	return &csipbv1.NodeStageVolumeRequest{
+		VolumeId:          r.ExternalID,
+		PublishContext:    r.PublishContext,
+		StagingTargetPath: r.StagingTargetPath,
+		VolumeCapability:  r.VolumeCapability.ToCSIRepresentation(),
+		Secrets:           r.Secrets,
+		VolumeContext:     r.VolumeContext,
+	}
+}
+
+func (r *NodeStageVolumeRequest) Validate() error {
```

Comment:
Was the intent to use this helper instead of the manual checks in `NodeStageVolume` ?

Reply:
Oops, forgot to clean that up. Will fix.

Reply:
Done in 139bba1